### PR TITLE
migrate specific Transaction and transaction bottom sheet to compose

### PR DESCRIPTION
**Pull request by AvneetSingh2001**
This PR contains migration of Two screens, as they both were dependent on each other. 

Jira Task: [MW-27](https://mifosforge.jira.com/browse/MW-27?atlOrigin=eyJpIjoiOGRhNDhhMjAzOTBlNDM4ODk3MjY2ZmY5YmVkMDMyZDUiLCJwIjoiaiJ9)

Jira Task: [MW-28](https://mifosforge.jira.com/browse/MW-28?atlOrigin=eyJpIjoiNzE0ZmY1OGUzMjNiNGQyYmExZWNiZWIyNTRiNGRiOTciLCJwIjoiaiJ9)

[MW-27]: https://mifosforge.jira.com/browse/MW-27?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[MW-28]: https://mifosforge.jira.com/browse/MW-28?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mifospay/src/main/java/org/mifospay/history/specific_transactions/presenter/SpecificTransactionsViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+class SpecificTransactionsViewModel @Inject constructor(
+    private val mUseCaseFactory: UseCaseFactory,
+    private var mTaskLooper: TaskLooper? = null
+): ViewModel() {
+
+    private val _uiState: MutableStateFlow<SpecificTransactionsUiState> = MutableStateFlow(SpecificTransactionsUiState.Loading)
+    val uiState get() = _uiState
+
+    fun getSpecificTransactions(
+        transactions: ArrayList<Transaction?>?,
```
Comment (PratyushSingh07) [inline mifospay/src/main/java/org/mifospay/history/specific_transactions/presenter/SpecificTransactionsViewModel.kt]: i dont think Transaction will be null

File: mifospay/src/main/java/org/mifospay/history/specific_transactions/presenter/SpecificTransactionsViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        secondAccountNumber: String?
+    ): ArrayList<Transaction?> {
+
+        val specificTransactions = ArrayList<Transaction?>()
+        if (transactions?.size!! > 0) {
+            for (i in transactions.indices) {
+                val transaction = transactions[i]
+                if (transaction?.transferDetail != null ) {
+                    val transferId = 2L
```
Comment (PratyushSingh07) [inline mifospay/src/main/java/org/mifospay/history/specific_transactions/presenter/SpecificTransactionsViewModel.kt]: add comment as to why it is hardcoded

File: mifospay/src/main/java/org/mifospay/history/specific_transactions/ui/SpecificTransactionsScreen.kt [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+    transactionItemClicked: (String) -> Unit,
+) {
+    MifosScaffold(
+        topBarTitle = R.string.specific_transactions_history,
+        backPress = backPress,
+        scaffoldContent = { paddingValues ->
+            Box(modifier = Modifier.padding(paddingValues)) {
+                when (uiState) {
+                    SpecificTransactionsUiState.Error -> {
+                        EmptyContentScreen(
```
Comment (PratyushSingh07) [inline mifospay/src/main/java/org/mifospay/history/specific_transactions/ui/SpecificTransactionsScreen.kt]: use error screen content

File: mifospay/src/main/java/org/mifospay/history/specific_transactions/ui/SpecificTransactionsScreen.kt [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+        }
+        Spacer(modifier = Modifier.height(10.dp))
+        Row(
+            modifier = Modifier.padding(horizontal = 10.dp),
+            horizontalArrangement = Arrangement.SpaceBetween,
+            verticalAlignment = Alignment.CenterVertically
+        ) {
+            Column {
+                Text(
+                    text = Constants.TRANSACTION_ID + ": " + transaction.transactionId,
```
Comment (PratyushSingh07) [inline mifospay/src/main/java/org/mifospay/history/specific_transactions/ui/SpecificTransactionsScreen.kt]: res

File: mifospay/src/main/java/org/mifospay/history/specific_transactions/ui/SpecificTransactionsScreen.kt [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+                    style = MaterialTheme.typography.bodyLarge,
+                    color = primaryDarkBlue
+                )
+                Text(
+                    text = Constants.DATE + ": " + transaction.date,
```
Comment (PratyushSingh07) [inline mifospay/src/main/java/org/mifospay/history/specific_transactions/ui/SpecificTransactionsScreen.kt]: ...

File: mifospay/src/main/java/org/mifospay/history/transaction_detail/ui/TransactionDetailScreen.kt [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+            }
+
+            is TransactionDetailUiState.Loading -> {
+                MfLoadingWheel(
+                    backgroundColor = Color.White
+                )
+            }
+
+            is TransactionDetailUiState.Success -> {
+
```
Comment (PratyushSingh07) [inline mifospay/src/main/java/org/mifospay/history/transaction_detail/ui/TransactionDetailScreen.kt]: ...

File: mifospay/src/main/java/org/mifospay/history/transaction_detail/ui/TransactionDetailScreen.kt [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+        Row(
+            modifier = Modifier
+                .padding(horizontal = 10.dp)
+                .fillMaxWidth(),
+            horizontalArrangement = Arrangement.SpaceBetween,
+            verticalAlignment = Alignment.CenterVertically
+        ) {
+            Column {
+                Text(
+                    text = Constants.TRANSACTION_ID + ": " + transaction.transactionId,
```
Comment (PratyushSingh07) [inline mifospay/src/main/java/org/mifospay/history/transaction_detail/ui/TransactionDetailScreen.kt]: res

File: mifospay/src/main/java/org/mifospay/history/transaction_detail/ui/TransactionDetailScreen.kt [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+fun TransactionDetailScreen(
+    uiState: TransactionDetailUiState,
+    transaction: Transaction,
+    viewReceipt: () -> Unit,
+    accountClicked: (String) -> Unit,
+) {
+    Box(modifier = Modifier.padding(20.dp).height(300.dp)) {
+        when (uiState) {
+            is TransactionDetailUiState.Error -> {
+                TransactionsDetailContent(
```
Comment (PratyushSingh07) [inline mifospay/src/main/java/org/mifospay/history/transaction_detail/ui/TransactionDetailScreen.kt]: replace with error content ?

**Review by PratyushSingh07 — CHANGES_REQUESTED** [review]

File: mifospay/src/main/java/org/mifospay/history/specific_transactions/presenter/SpecificTransactionsViewModel.kt [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+class SpecificTransactionsViewModel @Inject constructor(
+    private val mUseCaseFactory: UseCaseFactory,
+    private var mTaskLooper: TaskLooper? = null
+): ViewModel() {
+
+    private val _uiState: MutableStateFlow<SpecificTransactionsUiState> = MutableStateFlow(SpecificTransactionsUiState.Loading)
+    val uiState get() = _uiState
+
+    fun getSpecificTransactions(
+        transactions: ArrayList<Transaction>?,
```
Comment (Aditya-gupta99) [inline mifospay/src/main/java/org/mifospay/history/specific_transactions/presenter/SpecificTransactionsViewModel.kt]: Using nullable in transactions makes no sense

File: mifospay/src/main/java/org/mifospay/history/specific_transactions/ui/SpecificTransactionsScreen.kt [hunk 10]
```diff
@@ -0,0 +1,263 @@
...
+import org.mifospay.history.specific_transactions.presenter.SpecificTransactionsUiState
+import org.mifospay.history.specific_transactions.presenter.SpecificTransactionsViewModel
+import org.mifospay.theme.MifosTheme
+
+
+@Composable
+fun SpecificTransactionsScreen(
+    viewModel: SpecificTransactionsViewModel = hiltViewModel(),
+    transactions: ArrayList<Transaction>,
+    secondAccountNumber: String?,
```
Comment (Aditya-gupta99) [inline mifospay/src/main/java/org/mifospay/history/specific_transactions/ui/SpecificTransactionsScreen.kt]: You can remove this nullable

**Review by Aditya-gupta99 — CHANGES_REQUESTED** [review]

File: mifospay/src/main/java/org/mifospay/history/specific_transactions/ui/SpecificTransactionsScreen.kt [hunk 11]
```diff
@@ -0,0 +1,261 @@
...
+    modifier: Modifier = Modifier
+) {
+    Column(modifier = modifier.padding(horizontal = 10.dp)) {
+        Row(verticalAlignment = Alignment.CenterVertically) {
+            SpecificTransactionAccountInfo(
+                modifier = Modifier.weight(1f),
+                account = transaction.transferDetail.fromAccount,
+                client = transaction.transferDetail.fromClient
+            )
+            Image(painter = painterResource(id = R.drawable.ic_send), contentDescription = null)
```
Comment (therajanmaurya) [inline mifospay/src/main/java/org/mifospay/history/specific_transactions/ui/SpecificTransactionsScreen.kt]: Move icon in MifosIcons and use it here

File: mifospay/src/main/java/org/mifospay/history/specific_transactions/ui/SpecificTransactionsScreen.kt [hunk 12]
```diff
@@ -0,0 +1,261 @@
...
+                )
+                Text(
+                    text = stringResource(id = R.string.date) + transaction.date,
+                    style = MaterialTheme.typography.bodyLarge,
+                )
+                Text(
+                    text = when (transaction.transactionType) {
+                        TransactionType.DEBIT -> Constants.DEBIT
+                        TransactionType.CREDIT -> Constants.CREDIT
+                        TransactionType.OTHER -> Constants.OTHER
```
Comment (therajanmaurya) [inline mifospay/src/main/java/org/mifospay/history/specific_transactions/ui/SpecificTransactionsScreen.kt]: move this in strings.xml

File: mifospay/src/main/java/org/mifospay/history/specific_transactions/ui/SpecificTransactionsScreen.kt [hunk 13]
```diff
@@ -0,0 +1,261 @@
...
+                    style = MaterialTheme.typography.bodyLarge,
+                )
+            }
+            Spacer(modifier = Modifier.weight(1f))
+            Text(
+                text = "${transaction.currency.code}${transaction.amount}",
+                style = MaterialTheme.typography.displaySmall,
+                color = when (transaction.transactionType) {
+                    TransactionType.DEBIT -> Color.Red
+                    TransactionType.CREDIT -> Color(0xFF009688)
```
Comment (therajanmaurya) [inline mifospay/src/main/java/org/mifospay/history/specific_transactions/ui/SpecificTransactionsScreen.kt]: No hard coded color

**Review by therajanmaurya — CHANGES_REQUESTED** [review]

File: mifospay/src/main/java/org/mifospay/history/specific_transactions/presenter/SpecificTransactionsViewModel.kt (hunk 1 above)
Comment (AvneetSingh2001) [reply mifospay/src/main/java/org/mifospay/history/specific_transactions/presenter/SpecificTransactionsViewModel.kt]: done

File: mifospay/src/main/java/org/mifospay/history/specific_transactions/presenter/SpecificTransactionsViewModel.kt (hunk 2 above)
Comment (AvneetSingh2001) [reply mifospay/src/main/java/org/mifospay/history/specific_transactions/presenter/SpecificTransactionsViewModel.kt]: done

File: mifospay/src/main/java/org/mifospay/history/specific_transactions/ui/SpecificTransactionsScreen.kt (hunk 3 above)
Comment (AvneetSingh2001) [reply mifospay/src/main/java/org/mifospay/history/specific_transactions/ui/SpecificTransactionsScreen.kt]: done

File: mifospay/src/main/java/org/mifospay/history/specific_transactions/ui/SpecificTransactionsScreen.kt (hunk 4 above)
Comment (AvneetSingh2001) [reply mifospay/src/main/java/org/mifospay/history/specific_transactions/ui/SpecificTransactionsScreen.kt]: done

File: mifospay/src/main/java/org/mifospay/history/specific_transactions/ui/SpecificTransactionsScreen.kt (hunk 5 above)
Comment (AvneetSingh2001) [reply mifospay/src/main/java/org/mifospay/history/specific_transactions/ui/SpecificTransactionsScreen.kt]: done

File: mifospay/src/main/java/org/mifospay/history/transaction_detail/ui/TransactionDetailScreen.kt (hunk 6 above)
Comment (AvneetSingh2001) [reply mifospay/src/main/java/org/mifospay/history/transaction_detail/ui/TransactionDetailScreen.kt]: done

File: mifospay/src/main/java/org/mifospay/history/transaction_detail/ui/TransactionDetailScreen.kt (hunk 7 above)
Comment (AvneetSingh2001) [reply mifospay/src/main/java/org/mifospay/history/transaction_detail/ui/TransactionDetailScreen.kt]: done

File: mifospay/src/main/java/org/mifospay/history/transaction_detail/ui/TransactionDetailScreen.kt (hunk 8 above)
Comment (AvneetSingh2001) [reply mifospay/src/main/java/org/mifospay/history/transaction_detail/ui/TransactionDetailScreen.kt]: done

File: mifospay/src/main/java/org/mifospay/history/specific_transactions/presenter/SpecificTransactionsViewModel.kt (hunk 9 above)
Comment (AvneetSingh2001) [reply mifospay/src/main/java/org/mifospay/history/specific_transactions/presenter/SpecificTransactionsViewModel.kt]: done

File: mifospay/src/main/java/org/mifospay/history/specific_transactions/ui/SpecificTransactionsScreen.kt (hunk 10 above)
Comment (AvneetSingh2001) [reply mifospay/src/main/java/org/mifospay/history/specific_transactions/ui/SpecificTransactionsScreen.kt]: done

File: mifospay/src/main/java/org/mifospay/history/specific_transactions/ui/SpecificTransactionsScreen.kt (hunk 11 above)
Comment (AvneetSingh2001) [reply mifospay/src/main/java/org/mifospay/history/specific_transactions/ui/SpecificTransactionsScreen.kt]: done

File: mifospay/src/main/java/org/mifospay/history/specific_transactions/ui/SpecificTransactionsScreen.kt (hunk 12 above)
Comment (AvneetSingh2001) [reply mifospay/src/main/java/org/mifospay/history/specific_transactions/ui/SpecificTransactionsScreen.kt]: done

File: mifospay/src/main/java/org/mifospay/history/specific_transactions/ui/SpecificTransactionsScreen.kt (hunk 13 above)
Comment (AvneetSingh2001) [reply mifospay/src/main/java/org/mifospay/history/specific_transactions/ui/SpecificTransactionsScreen.kt]: done

**Pull request merged**